src/main/java/Duke/Storage.java

```diff
@@ -0,0 +1,132 @@
+     */
+    public static void saveList(ArrayList<Task> userLists) {
+        String pathName = "./data/duke.txt";
+        File file = new File(pathName);
+        try {
+            if (file.createNewFile()) {
+                //new file created
+                writeToFile(pathName, userLists);
+            } else {
+                //file has already been created
+                writeToFile(pathName, userLists);
+            }
+        } catch (IOException e) {
+            File directory = new File("./data/");
+            boolean isDirCreated = directory.mkdir();
+            if (isDirCreated) {
+                saveList(userLists);
+            }
+        }
+    }
+
+    /**
+     * Write the list of user tasks to file specified
+     * by the path
+     *
+     * @param pathName string with the path of file
+     * @param userList Arraylist of user tasks
+     * @throws IOException Unable to write to file
+     */
+    public static void writeToFile(String pathName, ArrayList<Task> userList) throws IOException {
+        FileWriter fileWriter = new FileWriter(pathName);
+        fileWriter.write(TaskList.listTask(userList));
+        fileWriter.close();
+    }
+
+    /**
+     * Load the file duke.txt and parse the strings
+     * inside the file to add those tasks back to
+     * the ArrayList of user tasks
+     *
+     * @param userLists ArrayList of user tasks
+     */
+    public static void loadSaveFile(ArrayList<Task> userLists) {
+        String pathName = "./data/duke.txt";
+        File file = new File(pathName);
+        try {
+            if (file.exists()) {
+                //save file exists, load in save file
+                Scanner fileContent = new Scanner(file);
+                while (fileContent.hasNext()) {
+                    String listContent = fileContent.nextLine();
+                    String regexTask = "(?<task>\\[[TDE]])";
+                    Matcher matcher =  regexMatching(regexTask, listContent);
+                    if (!matcher.find()) {
+                        return;
+                    }
+                    String task = matcher.group("task");
+                    switch (task) {
+                    case "[T]":
+                        String regexTodo = "(?<task>\\[[TDE]])(?<mark>\\[[\\s|X]])"+
+                                "(?<description>\\D*)";
+                        Matcher matcherTodo =  regexMatching(regexTodo, listContent);
+                        matcherTodo.find();
+                        String description = matcherTodo.group("description").trim();
+                        Todo newTodo = new Todo(description);
+                        if (matcherTodo.group("mark").equals("[X]")) {
+                            newTodo.setMark();
+                        }
+                        userLists.add(newTodo);
+                        break;
+                    case "[D]":
+                        String regexDeadline = "(?<task>\\[[TDE]])(?<mark>\\[[\\s|X]])"+
```

Comment:
The same code block is used for event case. How about using same methods for deadline and event, and instantiating the new objects separately? This will satisfy SLAP.